pilot/pkg/model/sidecar.go

```diff
@@ -124,6 +128,16 @@ type IstioEgressListenerWrapper struct {
 	virtualServices []Config
 }
 
+func createNamespaceForHostname(egress []*IstioEgressListenerWrapper) map[Hostname]string {
+	var namespaceForHostname = make(map[Hostname]string)
+	for _, egress := range egress {
+		for _, svc := range egress.Services() {
+			namespaceForHostname[svc.Hostname] = svc.Attributes.Namespace
```

Comment:
dont you need an existence check to prevent overwrites? services are stored in order of creation IIRC